third_party/txt/src/skia/paragraph_skia.cc

```diff
@@ -16,40 +16,137 @@
 
 #include "paragraph_skia.h"
 
+#include <algorithm>
+
 namespace txt {
 
 namespace skt = skia::textlayout;
 
+namespace {
+
+// Convert SkFontStyle::Weight values (ranging from 100-900) to txt::FontWeight
+// values (ranging from 0-8).
+txt::FontWeight GetTxtFontWeight(int font_weight) {
+  int txt_weight = (font_weight - 100) / 100;
+  txt_weight = std::clamp(txt_weight, static_cast<int>(txt::FontWeight::w100),
+                          static_cast<int>(txt::FontWeight::w900));
+  return static_cast<txt::FontWeight>(txt_weight);
+}
+
+txt::FontStyle GetTxtFontStyle(SkFontStyle::Slant font_slant) {
+  return font_slant == SkFontStyle::Slant::kUpright_Slant
+             ? txt::FontStyle::normal
+             : txt::FontStyle::italic;
+}
+
+TextStyle SkiaToTxt(const skt::TextStyle& skia) {
```

Comment:
A bit more of a descriptive name such as `SkiaTextStyleToTxt` would probably make more sense.